model/core/sys_regs.sail

```diff
@@ -288,15 +276,19 @@ function clause write_CSR((0x310, value) if xlen == 32) = { mstatus = legalize_m
 // architecture and extension checks
 
 function cur_architecture() -> Architecture = {
-  let a : arch_xlen =
-    match cur_privilege {
-      Machine           => misa[MXL],
-      Supervisor        => get_mstatus_SXL(mstatus),
-      User              => get_mstatus_UXL(mstatus),
-      VirtualUser       => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
-      VirtualSupervisor => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
-    };
-  architecture(a)
+  if   xlen == 32
+  then RV32
+  else {
+    let a : arch_xlen =
+      match cur_privilege {
+        Machine    => misa[MXL],
+        Supervisor => mstatus[SXL],
+        User       => mstatus[UXL],
+        VirtualUser       => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
+        VirtualSupervisor => internal_error(__FILE__, __LINE__, "Hypervisor extension not supported"),
+      };
+    architecture(a)
```

Comment:
We could inline this call into each of the match clauses to reduce nesting by one level, but I don't mind either way.